### ISSUE TITLE
Exit Horizon if DB has not been migrated to required version

### ISSUE THREAD

**Issue by gituser** [stellar/go#778]
Why horizon isn't auto-migrating DB schema when you start a new version?

E.g. when you start `v0.15.2` and previously you had `v0.14.2` it will fail on updating history until you do manually `./horizon db migrate up`.

I don't get why you do not build migration mechanism into the daemon itself or do not start horizon at all (signaling that you need to migrate DB schema) if DB is not in an actual state?

Thanks.

**Comment by bartekn** [comment]
Some migrations can take a longer period of time to be applied (ex. adding a column or index on a big table). It's better to have a control when you want to run migrations.

**Comment by gituser** [comment]
I don't think you got correctly what I've reported here.

To be more specific here is an example:

Let's say I had installed `version A` and decided to upgrade to `version B`, I've replaced the binary and launched it without even looking into the logs at all and discovered after some time that it needed manual migration. 

`version B` shouldn't run at all if the DB scheme is not up to date don't you think asking to run a migration?
